lib/ansible/modules/network/nxos/nxos_igmp_interface.py

```diff
@@ -115,85 +118,68 @@
               device does not send group-specific queries.
               The default is disabled.
         type: bool
-        default: 'no'
     oif_routemap:
         description:
-            - Configure a routemap for static outgoing interface (OIF).
+            - Configure a routemap for static outgoing interface (OIF) or
+              keyword 'default'.
     oif_prefix:
         description:
-            - Configure a prefix for static outgoing interface (OIF).
+            - Configure a prefix for static outgoing interface (OIF). This
+              argument is deprecated, please use oif_ps instead.
     oif_source:
         description:
-            - Configure a source for static outgoing interface (OIF).
+            - Configure a source for static outgoing interface (OIF). This
+              argument is deprecated, please use oif_ps instead.
+    oif_ps:
+        description:
+            - Configure prefixes and sources for static outgoing interface (OIF). This
+              is a list of dict where each dict has source and prefix defined or just
+              prefix if source is not needed. The specified values will be configured
+              on the device and if any previous prefix/sources exist, they will be removed.
+              Keyword 'default' is also accpted which removes all existing prefix/sources.
+        version_added: 2.6
     restart:
         description:
-            - Restart IGMP.
+            - Restart IGMP. This is NOT idempotent as this is action only.
         type: bool
+        default: False
     state:
         description:
             - Manages desired state of the resource.
         default: present
-        choices: ['present', 'default']
+        choices: ['present', 'absent', 'default']
 '''
 EXAMPLES = '''
 - nxos_igmp_interface:
     interface: ethernet1/32
     startup_query_interval: 30
+    oif_ps: [{'prefix': '238.2.2.6'}, {'source': '1.1.1.1', 'prefix': '238.2.2.5'}]
```

Comment:
The ansible convention to use this as:
``` 
oif_ps:
  - { 'prefix': '238.2.2.6' }
  - { 'source': '1.1.1.1', 'prefix': '238.2.2.5' }
```
This will be considered as `list` as the type of the param is list.

Reply:
Made change here as well as the test file too.